src/parser/expressions.rs

```diff
@@ -40,26 +42,71 @@ pub fn check_balanced_brackets(expr: &str) -> Result<(), String> {
     Ok(())
 }
 
-/// A wrapper type for a regex literal so that we can implement PartialEq
 #[derive(Debug, Clone)]
-pub struct RegexLiteral(regex::Regex);
+pub struct RegexLiteral {
+    regex: Regex,
+    pattern: String, // Store the original pattern string for comparisons
+}
 
 impl RegexLiteral {
-    pub(super) fn new(regex: regex::Regex) -> Self {
-        Self(regex)
+    /// Create a new `RegexLiteral` with optional case-insensitive and multiline flags.
+    pub fn new(
+        pattern: &str,
+        case_insensitive: bool,
+        multi_line: bool,
+    ) -> Result<Self, regress::Error> {
+        // Add flags to the pattern string as needed
+        let mut adjusted_pattern = String::new();
+        if case_insensitive {
+            adjusted_pattern.push_str("(?i)");
+        }
+        if multi_line {
+            adjusted_pattern.push_str("(?m)");
+        }
+        adjusted_pattern.push_str(pattern);
+
+        let regex = Regex::new(&adjusted_pattern)?;
+
+        Ok(Self {
+            regex,
+            pattern: pattern.to_string(),
+        })
```

Comment:
Looks like this syntax isn't quite correct with `regress`. Using the following allows the old `regex_with_flags` test in `tokenizer.rs` to pass:

```suggestion
        // Add flags to the pattern string as needed
        let mut flags = String::new();
        if case_insensitive {
            flags.push('i');
        }
        if multi_line {
            flags.push('m');
        }

        let regex = Regex::with_flags(pattern, flags.as_str())?;

        Ok(Self {
            regex,
            pattern: pattern.to_string(),
        })
```